pkg/streamingpromql/types/ring_buffer_test.go

```diff
@@ -563,3 +563,363 @@ func setupRingBufferTestingPools(t *testing.T) {
+		t.Run(name, func(t *testing.T) {
+			buff := NewFPointRingBuffer(limiter.NewMemoryConsumptionTracker(context.Background(), 0, nil, ""))
+			require.NoError(t, buff.Use(tc.buff))
+			err := buff.ReplaceTail(tc.tail)
+			if len(tc.err) > 0 {
+				require.ErrorContains(t, err, tc.err)
+				return
+			}
+			require.NoError(t, err)
+			bufferMembersEquals(t, tc.expected, buff)
+		})
+	}
+}
+
+func TestFPointRingBufferReplaceValueAtPos(t *testing.T) {
+	testCases := map[string]struct {
+		position int
+		value    float64
+		err      string
+		buff     []promql.FPoint
+		expected []promql.FPoint
+	}{
+		"empty buff": {
+			position: 0,
+			value:    10,
+			buff:     []promql.FPoint{},
+			err:      "unable to replace value at position - position exceeds buffer size",
+		},
+		"position exceeds buffer size": {
+			position: 1,
+			value:    10,
+			buff:     []promql.FPoint{{T: 10, F: 20}},
+			err:      "unable to replace value at position - position exceeds buffer size",
+		},
+		"single point": {
+			position: 0,
+			value:    30,
+			buff:     []promql.FPoint{{T: 10, F: 20}},
+			expected: []promql.FPoint{{T: 10, F: 30}},
+		},
+		"multiple points - head": {
+			position: 0,
+			value:    30,
+			buff:     []promql.FPoint{{T: 10, F: 20}, {T: 20, F: 20}, {T: 30, F: 20}, {T: 40, F: 20}, {T: 50, F: 20}, {T: 60, F: 20}, {T: 70, F: 20}, {T: 80, F: 20}},
+			expected: []promql.FPoint{{T: 10, F: 30}, {T: 20, F: 20}, {T: 30, F: 20}, {T: 40, F: 20}, {T: 50, F: 20}, {T: 60, F: 20}, {T: 70, F: 20}, {T: 80, F: 20}},
+		},
+		"multiple points - tail": {
+			position: 7,
+			value:    30,
+			buff:     []promql.FPoint{{T: 10, F: 20}, {T: 20, F: 20}, {T: 30, F: 20}, {T: 40, F: 20}, {T: 50, F: 20}, {T: 60, F: 20}, {T: 70, F: 20}, {T: 80, F: 20}},
+			expected: []promql.FPoint{{T: 10, F: 20}, {T: 20, F: 20}, {T: 30, F: 20}, {T: 40, F: 20}, {T: 50, F: 20}, {T: 60, F: 20}, {T: 70, F: 20}, {T: 80, F: 30}},
+		},
+		"multiple points - mid": {
+			position: 3,
+			value:    30,
+			buff:     []promql.FPoint{{T: 10, F: 20}, {T: 20, F: 20}, {T: 30, F: 20}, {T: 40, F: 20}, {T: 50, F: 20}, {T: 60, F: 20}, {T: 70, F: 20}, {T: 80, F: 20}},
+			expected: []promql.FPoint{{T: 10, F: 20}, {T: 20, F: 20}, {T: 30, F: 20}, {T: 40, F: 30}, {T: 50, F: 20}, {T: 60, F: 20}, {T: 70, F: 20}, {T: 80, F: 20}},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			buff := NewFPointRingBuffer(limiter.NewMemoryConsumptionTracker(context.Background(), 0, nil, ""))
+			require.NoError(t, buff.Use(tc.buff))
+			err := buff.ReplaceValueAtPos(tc.position, tc.value)
+			if len(tc.err) > 0 {
+				require.ErrorContains(t, err, tc.err)
+				return
+			}
+			require.NoError(t, err)
+			bufferMembersEquals(t, tc.expected, buff)
+		})
+	}
+}
+
+func TestFPointRingBufferAppendSlice(t *testing.T) {
+
+	testCases := map[string]struct {
+		append   []promql.FPoint
+		buff     []promql.FPoint
+		expected []promql.FPoint
+		err      string
+	}{
+		"empty buff - nil slice": {
```

Comment:
[nit] In all of these test names (here and elsewhere), I would use the full word "buffer"